packages/react-components/src/components/xy-output-component.tsx

```diff
@@ -273,21 +283,125 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
         this.setState({orderedNodes: ids});
     }
 
-    private beginSelection(event: React.MouseEvent<HTMLDivElement, MouseEvent>) {
+    private onMouseDown(event: React.MouseEvent<HTMLDivElement, MouseEvent>) {
         this.mouseIsDown = true;
         this.posPixelSelect = event.nativeEvent.screenX;
-        const offset = this.props.viewRange.getOffset() ?? 0;
-        const scale = this.props.viewRange.getEnd() - this.props.viewRange.getstart();
-        const xPos = this.props.viewRange.getstart() - offset +
-            (event.nativeEvent.offsetX / this.lineChartRef.current.chartInstance.width) * scale;
+        const startTime = this.getTimeX(event.nativeEvent.offsetX);
         this.props.unitController.selectionRange = {
-            start: xPos,
-            end: xPos
+            start: startTime,
+            end: startTime
         };
         document.addEventListener('mousemove', this.updateSelection);
         document.addEventListener('mouseup', this.endSelection);
     }
 
+    private updateRange(rangeStart: number, rangeEnd: number) {
+        if (rangeEnd < rangeStart) {
+            const temp = rangeStart;
+            rangeStart = rangeEnd;
+            rangeEnd = temp;
+        }
+        this.props.unitController.viewRange = {
+            start: rangeStart,
+            end: rangeEnd
+        };
+    }
+
+    private zoom(iszoomIn: boolean) {
+        if (this.props.unitController.viewRangeLength >= 1) {
+            let newStartRange = 0;
+            let newEndRange = 0;
+            const zoomNumber = iszoomIn ? 1 : -1;
+                const position = this.getTimeX(this.positionXMove);
+                const startPos = position - this.props.unitController.viewRange.start;
+                const endPos = this.props.unitController.viewRange.end - position;
+                newStartRange = position - (startPos - zoomNumber * (startPos * TEN_PERCENT));
+                newEndRange = position + (endPos - zoomNumber * (endPos * TEN_PERCENT));
```

Comment:
Seems like the end result of this code is that range is changed by a factor of 0.9 or 1.1.
But to be the mathematical inverse, the factor should be 0.9 or 1/0.9=1.11111111111.